src/main/java/org/jabref/gui/groups/GroupDialog.java

```diff
@@ -438,11 +445,15 @@ private static String formatRegExException(String regExp, Exception e) {
         return s;
     }
 
-    private VBox createOptionsTexGroup() {
+    private VBox createOptionsTexGroup(JabRefFrame jabRefFrame) {
```

Comment:
Sorry, if I was a bit unspecific. The idea would be to remove the dependency on JabRefFrame completey;
a) pass DialogService in the constructor and b) an instance of JabRefPreferenecs
This should eliminate the need for the whole jabrefFrame object

Reply:
I understand what you mean now.
Thank you for the feedback, I'll get to it.

Reply:
The idea behind that is the following:  Only pass the absolutely needed objects (the more concrete the better) to remove unnecessary dependencies and to reduce potential side effects.

Reply:
`public GroupDialog(JabRefFrame jabrefFrame, AbstractGroup editedGroup)`
A JabRefFrame object is already being passed as a constructor argument. I was planning on defining an attribute DialogService that could be initialized in the constructor, something like `private final DialogService dialogService;`
This will remove the need to pass JabRefFrame object in function calls, as I had done before.
Would that resolve the issue?

Reply:
From what I saw it should be sufficient to pass a DialogService object and a JabRef Preferences object and of course your solution in addition 
Because that one other usage is the call to frame.prefs().... Can be replaced

Reply:
to make it more precise something like this:
```
public GroupDialog(DialogService dialogService, JabRefPrefernces prefs)
this.dialogService = dialogService;
this.prefs = prefs;
```

Reply:
If the **caller** of the dialog does not have a Preferences object defined, you can use `Globals.prefs` as parameter for the dialog. (This can be further refactored, but is a bit out of scope now for this PR)

I hope this doesn't confuse you now, just ask if you are not sure on how to proceed.

Reply:
Thank you for all the suggestions. I have modified the constructors to pass dialogService and prefs reference.

However, there was another usage of jabrefFrame in order to set the value of `Optional<GroupTreeNode> rootGroup=jabrefFrame.getCurrentBasePanel().getBibDatabaseContext().getMetaData().getGroups();`

Instead of this I am planning of getting the BasePanel object by `JabRefGUI.getMainFrame().getCurrentBasePanel()`
Would this be sufficient?

Also, as of now when we click on add subroup option, it shows Edit Group as the Title of the Dialog Box, wouldn't it be better if it showed Add Subgroup as the title in that case?

Reply:
> Instead of this I am planning of getting the BasePanel object by JabRefGUI.getMainFrame().getCurrentBasePanel()
Would this be sufficient?

Yes. It's not perfect since we also want to get rid of `JabRefGUI` too, but I don't see a better way. 

>Also, as of now when we click on add subroup option, it shows Edit Group as the Title of the Dialog Box, wouldn't it be better if it showed Add Subgroup as the title in that case?

Good suggestion! You may go ahead and implement it (but maybe as a separate PR to keep things simple).